source/jormungandr/jormungandr/scenarios/ridesharing/ridesharing_service.py

```diff
@@ -44,12 +47,30 @@ def status(self):
         """
         pass
 
+    def request_journeys_with_feed_publisher(self, from_coord, to_coord, period_extremity, limit=None):
+        """
+        This function shouldn't be overwritten!
+
+        :return: a list(mandatory) contains solutions and a feed_publisher
+        """
+        journeys = self._request_journeys(from_coord, to_coord, period_extremity, limit)
+        feed_publisher = self._get_feed_publisher()
+        return journeys, feed_publisher
+
+    @abc.abstractmethod
+    def _request_journeys(self, from_coord, to_coord, period_extremity, limit=None):
+        """
+        :return: a list(mandatory) contains solutions
+        """
+        pass
```

Comment:
I'd rather have the `return []`

Reply:
this function can never be called alone since it's abstract and MUST be implemented(like c++) or the python interpreter won't `compile`

the convention is either `pass` or `raise NotImplementedError` whatever the function should do in the real implementation

Reply:
My idea was to "explain" that it should absolutly be a list. Anyway, it's minor.